drivers/platform/aducm3029/spi.c

```diff
@@ -0,0 +1,228 @@
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "spi_extra.h"
+#include "spi.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/*****************************  Variables   **********************************/
+/******************************************************************************/
+
+static uint8_t	active_spi = 0; //Stores the active SPIs
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Allocates the memory needed for the descriptor
+ * @return A address to the allocated memory, NULL if the allocation fails.
+ */
+struct spi_desc *alloc_data_mem(void)
+{
+	struct spi_desc		*desc;
+	struct adicup_spi_desc	*adicup_desc;
+
+	desc = calloc(1, sizeof(struct spi_desc));
+	if (!desc)
+		return NULL;
+
+	adicup_desc = calloc(1, sizeof(struct adicup_spi_desc));
+	if (!adicup_desc) {
+		free(desc);
+		return NULL;
+	}
+	desc->extra = adicup_desc;
+
+	adicup_desc->buffer = calloc(1, ADI_SPI_MEMORY_SIZE);
+	if (NULL == adicup_desc->buffer) {
+		free(adicup_desc);
+		free(desc);
+		return NULL;
+	}
+
+	return desc;
+}
+
+/**
+ * @brief Deallocates the memory needed for the descriptor
+ * @param desc - Descriptor to be deallocated
+ */
+void free_desc_mem(struct spi_desc *desc)
+{
+	free(((struct adicup_spi_desc*)(desc->extra))->buffer);
+	free(desc->extra);
+	free(desc);
+}
+
+/**
+ * @brief Initialize the SPI communication peripheral.
+ * @param desc - The SPI descriptor.
+ * @param init_param - The structure that contains the SPI parameters.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t spi_init(struct spi_desc **desc,
+		 const struct spi_init_param *param)
+{
+	ADI_SPI_RESULT			spi_ret;
+	struct adicup_spi_desc		*adicup_desc;
+	struct adicup_spi_init_param	*adicup_init_param;
+
+
+	if (!desc || !param || !(param->extra))
+		return FAILURE;
+
+	/* Verify if the SPI is already activated */
+	adicup_init_param = param->extra;
+	if (active_spi & (1 << adicup_init_param->spi_channel))
```

Comment:
`adi_spi_Open` returns specific error code in case the spi channel is already in use. Do we still need this check considering this?

Reply:
Adi_spi_open returns already in use only in debug mode, so it is better to keep it